tools/flang2/flang2exe/cgmain.cpp

```diff
@@ -1769,6 +1788,24 @@ schedule(void)
             i->misc_metadata = loop_md;
           }
         }
+
+        if (BIH_VECTORLENGTH_ENABLED(bih)) {
+          if (LL_MDREF_IS_NULL(loop_md)) {
+            loop_md = cons_loop_id_md();
+          }
+          LL_MDRef vectorlength_enable = cons_vectorize_metadata();
+          ll_extend_md_node(cpu_llvm_module, loop_md, vectorlength_enable);
+          LL_MDRef vectorlength_scalable  = cons_vectorlength_scalable_metadata(BIH_VECTORLENGTH_SCALABLE( bih )>0);
```

Comment:
I don't think you need to compare this with 0, the value you extract is going to be suitable to pass to the function.